app/react-graphql/hooks/useQueryOne.tsx

```diff
@@ -40,22 +40,31 @@ export default function useQueryOne<
   });
 
   useEffect(() => {
-    if (item) {
-      setKey(keyExtractor(sharedConfig, item));
-    }
-  }, [item]);
+    updateItemKey();
+  }, [item, objectVariables]);
 
   useEffect(() => {
+    if (mutationEvent.type === 'init') {
+      return;
+    }
+
+    logMutationEvent();
+
     if (
-      mutationEvent.listKey == sharedConfig.typename &&
-      mutationEvent.key === key &&
-      mutationEvent.type !== 'init'
+      mutationEvent.listKey === sharedConfig.typename &&
+      (mutationEvent.key === key || !key)
     ) {
       if (mutationEvent.type === 'delete') {
-        setItem(undefined);
+        setItem(null);
       } else {
         setItem(mutationEvent.payload as TData);
       }
+    } else if (
+      mutationEvent.listKey &&
+      sharedConfig.typename &&
+      mutationEvent.listKey !== sharedConfig.typename
```

Comment:
Isn't this just for debugging? Should it be taken out now?

Reply:
My opinion is that if we're going to open this up, then providing intelligent and helpful console logs for errors is a feature.